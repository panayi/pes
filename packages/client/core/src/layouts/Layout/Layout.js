import React from 'react';
import PropTypes from 'prop-types';
import * as R from 'ramda';
import { isNotNil } from 'ramda-adjunct';
import classNames from 'classnames';
import { withProps } from 'recompose';
import withStyles from '@material-ui/core/styles/withStyles';
import ReduxModal from '../../components/Modal/ReduxModal/ReduxModal';
import Login from '../../modules/Login/Login';
import Header from './Header/Header';
import Sidebar from './Sidebar/Sidebar';
import Page from './Page/Page';

const styles = theme => ({
  '@global': {
    body: {
      backgroundColor: '#f0f1f2',
    },
    'html,body,#root,#root2,#root > div,#root2 > div': {
      height: '100%',
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    minHeight: '100%',
    width: '100%',
  },
  hasHeader: {
    marginTop: theme.layout.headerHeight.tablet,
    [theme.breakpoints.down(theme.map.phone)]: {
      marginTop: theme.layout.headerHeight.phone,
    },
  },
});

const Layout = ({
  header,
  sidebar,
  children,
  pageClassName,
  headerClasses,
  sidebarClassName,
  fixed,
  flex,
  wide,
  hasHeader,
  hasSidebar,
  classes,
}) => (
  <div className={classes.root}>
    {hasHeader && <Header classes={headerClasses}>{header}</Header>}
    {hasSidebar && <Sidebar className={sidebarClassName}>{sidebar}</Sidebar>}
    <Page
      className={classNames(pageClassName, {
        [classes.hasHeader]: hasHeader,
      })}
      fixed={fixed}
      flex={flex}
      wide={wide}
    >
      {children}
    </Page>
    <ReduxModal id="login" content={Login} />
  </div>
);

Layout.propTypes = {
  header: PropTypes.node,
  sidebar: PropTypes.node,
  children: PropTypes.node,
  pageClassName: PropTypes.string,
  fixed: PropTypes.bool,
  flex: PropTypes.bool,
  wide: PropTypes.bool,
  classes: PropTypes.shape({}).isRequired,
};

Layout.defaultProps = {
  header: null,
  sidebar: null,
  children: null,
  pageClassName: null,
  fixed: false,
  flex: false,
  wide: false,
};

export default R.compose(
  withProps(({ header, sidebar }) => ({
    hasHeader: isNotNil(header),
    hasSidebar: isNotNil(sidebar),
  })),
  withStyles(styles),
)(Layout);
