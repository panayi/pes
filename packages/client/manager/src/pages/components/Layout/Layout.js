import React from 'react';
import PropTypes from 'prop-types';
import * as R from 'ramda';
import classNames from 'classnames';
import { withProps } from 'recompose';
import { createStructuredSelector } from 'reselect';
import Badge from '@material-ui/core/Badge';
import Drawer from '@material-ui/core/Drawer';
import IconButton from '@material-ui/core/IconButton';
import withStyles from '@material-ui/core/styles/withStyles';
import AddIcon from '@material-ui/icons/Add';
import * as taskTypes from '@pesposa/core/src/config/taskTypes';
import ReduxModal from '@pesposa/client-core/src/components/Modal/ReduxModal/ReduxModal';
import { connectData } from '@pesposa/client-core/src/lib/connectData';
import { models } from '@pesposa/client-core/src/store/firebase/data';
import { actions as modalActions } from '@pesposa/client-core/src/store/modals';
import BaseLayout from '@pesposa/client-core/src/layouts/Layout/Layout';
import { selectors as drawerSelectors } from 'store/drawer';
import CreateSubmissionTask from 'modules/SubmissionTask/CreateSubmissionTask/CreateSubmissionTask';
import Header from './Header/Header';
import Sidebar from './Sidebar/Sidebar';
import ToggleDrawer from './ToggleDrawer/ToggleDrawer';

const CREATE_SUBMISSION_MODAL_ID = 'createSubmission';
const DRAWER_WIDTH = 250;
const TOOLBAR_HEIGHT = 48;

const styles = theme => ({
  '@global': {
    html: {
      background: '#fafafa',
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    width: '100%',
    height: '100%',
  },
  appBar: {
    background: theme.palette.secondary.dark,
    color: theme.palette.common.white,
  },
  toolbar: {
    borderBottom: 'none',
  },
  drawer: {
    position: 'relative',
    height: '100%',
    width: DRAWER_WIDTH,
  },
  drawerPaper: {
    width: DRAWER_WIDTH,
  },
  content: {
    display: 'flex',
    width: '100%',
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    [theme.breakpoints.up('sm')]: {
      content: {
        height: `calc(100% - ${TOOLBAR_HEIGHT}px)`,
        marginTop: TOOLBAR_HEIGHT,
      },
    },
  },
  contentLeft: {
    marginLeft: -DRAWER_WIDTH,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  contentShiftLeft: {
    marginLeft: 0,
  },
  createSubmissionModal: {
    maxWidth: 'none',
  },
  badge: {
    top: 0,
    right: -6,
  },
});

const Layout = props => {
  const {
    title,
    children,
    isDrawerOpened,
    noSidebar,
    tasksCount,
    openModal,
    classes,
  } = props;

  return (
    <BaseLayout
      headerClasses={{ appBar: classes.appBar, toolbar: classes.toolbar }}
      header={
        <Header
          title={title}
          actions={
            tasksCount.total > 0 ? (
              <Badge
                classes={{ badge: classes.badge }}
                badgeContent={tasksCount.total}
                color="primary"
              >
                <ToggleDrawer />
              </Badge>
            ) : (
              <ToggleDrawer />
            )
          }
          secondaryActions={
            <IconButton
              color="inherit"
              onClick={() => openModal(CREATE_SUBMISSION_MODAL_ID)}
            >
              <AddIcon />
            </IconButton>
          }
        />
      }
    >
      <div className={classes.root}>
        <Drawer
          variant="persistent"
          classes={{
            paper: classes.drawerPaper,
          }}
          className={classes.drawer}
          anchor="left"
          open={isDrawerOpened}
        >
          {!noSidebar && <Sidebar tasksCount={tasksCount} />}
        </Drawer>
        <main
          className={classNames(classes.content, classes.contentLeft, {
            [classes.contentShift]: isDrawerOpened,
            [classes.contentShiftLeft]: isDrawerOpened,
          })}
        >
          {children}
        </main>
      </div>
      <ReduxModal
        id={CREATE_SUBMISSION_MODAL_ID}
        maxWidth="md"
        content={CreateSubmissionTask}
      />
    </BaseLayout>
  );
};

Layout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  isDrawerOpened: PropTypes.bool.isRequired,
  classes: PropTypes.shape({}).isRequired,
};

Layout.defaultProps = {
  title: '',
  children: null,
};

const mapDataToProps = {
  submissionTasksCount: models.taskCounters(R.always(taskTypes.SUBMISSION)),
  reviewAdTasksCount: models.taskCounters(R.always(taskTypes.REVIEW_AD)),
  convertExternalUserTasksCount: models.taskCounters(
    R.always(taskTypes.CONVERT_EXTERNAL_USER),
  ),
};

const mapStateToProps = createStructuredSelector({
  isDrawerOpened: drawerSelectors.isOpenedSelector,
});

const mapDispatchToProps = {
  openModal: modalActions.openModal,
};

export default R.compose(
  connectData(mapDataToProps, mapStateToProps, mapDispatchToProps),
  withProps(
    ({
      submissionTasksCount,
      reviewAdTasksCount,
      convertExternalUserTasksCount,
    }) => {
      const finalSubmissionTasksCount = submissionTasksCount || 0;
      const finalReviewAdTasksCount = reviewAdTasksCount || 0;
      const finalConvertExternalUserTasksCount =
        convertExternalUserTasksCount || 0;

      return {
        tasksCount: {
          total:
            finalSubmissionTasksCount +
            finalReviewAdTasksCount +
            finalConvertExternalUserTasksCount,
          submission: finalSubmissionTasksCount,
          reviewAd: finalReviewAdTasksCount,
          convertExternalUser: finalConvertExternalUserTasksCount,
        },
      };
    },
  ),
  withStyles(styles),
)(Layout);
