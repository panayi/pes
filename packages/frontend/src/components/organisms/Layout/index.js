/* @flow */
import React from 'react';
import { Configure } from 'react-instantsearch/dom';
import { withStyles } from 'material-ui/styles';
import Header from 'components/organisms/Header';
import SearchAds from 'components/organisms/SearchAds';

type Props = {
  children: React$Node,
  classes: Object,
  configureSearch: React$Element<Configure> | null,
};

const styles = theme => ({
  '@global': {
    body: {
      marginTop: theme.layout.headerHeight,
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    // TODO: Use constant for header height
    minHeight: `calc(100vh - ${theme.layout.headerHeight}px)`,
    width: '100%',
  },
});

const Layout = ({ configureSearch, children, classes }: Props) => (
  <div className={classes.root}>
    <Header />
    {children}
    {configureSearch || <SearchAds.ConfigureSearchParams />}
  </div>
);

export default withStyles(styles)(Layout);
