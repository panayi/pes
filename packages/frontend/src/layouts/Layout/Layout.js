/* @flow */
import React from 'react';
import { withStyles } from 'material-ui/styles';
import Sidebar from './Sidebar/Sidebar';
import Page from './Page/Page';

type Props = {
  header: Class<React$Component<*>>,
  sidebar: Class<React$Component<*>>,
  children: React$Node,
  classes: Object,
  // Page props
  pageClassName: ?string,
  fixed: ?boolean,
  flex: ?boolean,
  wide: ?boolean,
};

const styles = theme => ({
  '@global': {
    body: {
      marginTop: theme.layout.headerHeight,
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    // TODO: Use constant for header height
    minHeight: `calc(100vh - ${theme.layout.headerHeight}px)`,
    width: '100%',
    backgroundColor: theme.palette.grey[200],
  },
});

const Layout = ({
  header: Header,
  sidebar: SidebarContent,
  children,
  classes,
  pageClassName,
  fixed,
  flex,
  wide,
}: Props) => (
  <div className={classes.root}>
    <Header />
    {SidebarContent ? (
      <Sidebar>
        <SidebarContent />
      </Sidebar>
    ) : null}
    <Page className={pageClassName} fixed={fixed} flex={flex} wide={wide}>
      {children}
    </Page>
  </div>
);

export default withStyles(styles)(Layout);
