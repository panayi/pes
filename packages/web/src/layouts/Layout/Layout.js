import React from 'react';
import PropTypes from 'prop-types';
import elementType from 'prop-types-extra/lib/elementType';
import * as R from 'ramda';
import classNames from 'classnames';
import { isNotNil } from 'ramda-adjunct';
import { withProps } from 'recompose';
import withStyles from 'material-ui/styles/withStyles';
import ReduxModal from 'components/Modal/ReduxModal/ReduxModal';
import Login from 'modules/Login/Login';
import Header from './Header/Header';
import Sidebar from './Sidebar/Sidebar';
import Page from './Page/Page';

const styles = theme => ({
  '@global': {
    body: {
      backgroundColor: '#f0f1f2',
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    minHeight: 'calc(100vh)',
    width: '100%',
  },
  hasHeader: {
    marginTop: theme.layout.headerHeight.phone,
    [theme.breakpoints.up(theme.map.tablet)]: {
      marginTop: theme.layout.headerHeight.tablet,
    },
  },
});

const Layout = ({
  header: HeaderContent,
  sidebar: SidebarContent,
  children,
  pageClassName,
  headerClassName,
  sidebarClassName,
  fixed,
  flex,
  wide,
  hasHeader,
  hasSidebar,
  classes,
}) => (
  <div className={classes.root}>
    {hasHeader && (
      <Header className={headerClassName}>
        <HeaderContent />
      </Header>
    )}
    {hasSidebar && (
      <Sidebar className={sidebarClassName}>
        <SidebarContent />
      </Sidebar>
    )}
    <Page
      className={classNames(pageClassName, { [classes.hasHeader]: hasHeader })}
      fixed={fixed}
      flex={flex}
      wide={wide}
    >
      {children}
    </Page>
    <ReduxModal id="login" content={Login} />
  </div>
);

Layout.propTypes = {
  header: elementType,
  sidebar: elementType,
  children: PropTypes.node,
  pageClassName: PropTypes.string,
  fixed: PropTypes.bool,
  flex: PropTypes.bool,
  wide: PropTypes.bool,
  classes: PropTypes.shape({}).isRequired,
};

Layout.defaultProps = {
  header: null,
  sidebar: null,
  children: null,
  pageClassName: null,
  fixed: false,
  flex: false,
  wide: false,
};

export default R.compose(
  withProps(({ header, sidebar }) => ({
    hasHeader: isNotNil(header),
    hasSidebar: isNotNil(sidebar),
  })),
  withStyles(styles),
)(Layout);
