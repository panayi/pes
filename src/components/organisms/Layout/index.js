/* @flow */
import React from 'react';
import { withStyles } from 'material-ui/styles';
import Header from 'components/organisms/Header';

type Props = {
  children: React$Node,
  classes: Object,
};

const styles = {
  '@global': {
    body: {
      marginTop: '64px',
    },
  },
  root: {
    display: 'flex',
    alignItems: 'stretch',
    minHeight: '100vh',
    width: '100%',
  },
};

const Layout = ({ children, classes }: Props) => (
  <div className={classes.root}>
    <Header />
    {children}
  </div>
);

export default withStyles(styles)(Layout);
