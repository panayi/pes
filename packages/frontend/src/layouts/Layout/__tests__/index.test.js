import React from 'react';
import Layout from '../Layout';

it('should render correctly', () => {
  const wrapper = shallow(
    <Layout>
      <span>I am a child</span>
      <span>I am another child</span>
    </Layout>,
  );
  expect(wrapper).toMatchSnapshot();
});
