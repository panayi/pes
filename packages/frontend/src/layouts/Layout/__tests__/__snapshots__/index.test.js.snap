// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Layout
  classes={
    Object {
      "root": "Layout-root-1",
    }
  }
>
  <span>
    I am a child
  </span>
  <span>
    I am another child
  </span>
</Layout>
`;
